examples/with-loading/components/Header.js

```diff
@@ -4,12 +4,15 @@ import Link from 'next/link'
 import NProgress from 'nprogress'
 import Router from 'next/router'
 
-Router.onRouteChangeStart = (url) => {
-  console.log(`Loading: ${url}`)
-  NProgress.start()
-}
-Router.onRouteChangeComplete = () => NProgress.done()
-Router.onRouteChangeError = () => NProgress.done()
+Router.ready(() => {
+  const events = Router.events
+  events.on('routeChangeStart', (url) => {
+    console.log(`Loading: ${url}`)
+    NProgress.start()
+  })
+  events.on('routeChangeComplete', () => NProgress.done())
+  events.on('routeChangeError', () => NProgress.done())
```

Comment:
Why not use Router.events three times? IMO a new var creates more complexity than helps.

Reply:
Agreed, updated sample.